stable/anchore-engine/templates/enterprise_feeds_configmap.yaml

```diff
@@ -94,54 +94,51 @@ data:
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.debianDriverEnabled | quote) }}
           ol:
-            enabled: {{ default "true" .Values.anchoreEnterpriseFeeds.olDriverEnabled }}
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.olDriverEnabled | quote) }}
           ubuntu:
-            enabled: {{ default "true" .Values.anchoreEnterpriseFeeds.ubuntuDriverEnabled }}
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.ubuntuDriverEnabled | quote) }}
           rhel:
-            enabled: {{ default "true" .Values.anchoreEnterpriseFeeds.rhelDriverEnabled }}
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.rhelDriverEnabled | quote) }}
           nvddb:
-            enabled: {{ default "false" .Values.anchoreEnterpriseFeeds.nvdDriverEnabled }}
+            enabled: {{ default "false" (.Values.anchoreEnterpriseFeeds.nvdDriverEnabled | quote) }}
           # npm and gem drivers are explicitly disabled out of the box
           npm:
-            enabled: {{ default "false" .Values.anchoreEnterpriseFeeds.npmDriverEnabled }}
+            enabled: {{ default "false" (.Values.anchoreEnterpriseFeeds.npmDriverEnabled | quote) }}
           gem:
             # rubygem data comes packaged as a PostgreSQL dump file. gem driver loads the pg dump and normalizes the data.
             # To enable gem driver comment the enabled property and uncomment the db_connect property.
-            enabled: {{ default "false" .Values.anchoreEnterpriseFeeds.gemDriverEnabled }}
+            enabled: {{ default "false" (.Values.anchoreEnterpriseFeeds.gemDriverEnabled | quote) }}
             db_connect: {{ default "'postgresql://${ANCHORE_DB_USER}:${ANCHORE_FEEDS_DB_PASSWORD}@${ANCHORE_DB_HOST}/gems'" .Values.anchoreEnterpriseFeeds.gemDbEndpoint }}
           nvdv2:
-            enabled: {{ default "true" .Values.anchoreEnterpriseFeeds.nvdv2DriverEnabled }}
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.nvdv2DriverEnabled | quote) }}
           vulndb:
-            enabled: {{ default "true" .Values.anchoreEnterpriseFeeds.vulndbDriverEnabled }}
-        {{- if .Values.anchoreEnterpriseFeeds.msrcDriverEnabled }}
+            enabled: {{ default "true" (.Values.anchoreEnterpriseFeeds.vulndbDriverEnabled | quote) }}
           msrc:
-            enabled: true
-            api_key: ${ANCHORE_MSRC_KEY}
+            enabled: {{ default "false" (.Values.anchoreEnterpriseFeeds.githubDriverEnabled | quote) }}
             {{- with .Values.anchoreEnterpriseFeeds.msrcWhitelist }}
             whitelist:
               - {{ . }}
             {{- end }}
-        {{- end }}
-        {{- if .Values.anchoreEnterpriseFeeds.githubDriverEnabled }}
           github:
-            enabled: true
+            enabled: {{ default "false" (.Values.anchoreEnterpriseFeeds.githubDriverEnabled | quote) }}
+          {{- if .Values.anchoreEnterpriseFeeds.githubDriverEnabled }}
```

Comment:
Should this not be `{{- if eq .Values.anchoreEnterpriseFeeds.githubDriverEnabled "true"}}`, since we are dealing in strings and not actual booleans?

Reply:
Tested with `helm template -g ./ -f ./engine-rc-values.yaml -s templates/enterprise_feeds_configmap.yaml`

Reply:
Good observation. You're right that we're comparing strings not booleans, however if the value in the values.yaml file isn't quoted, then you'll get an error - `error calling eq: incompatible types for comparison`. We could fix this by using:
```
{{- if eq (.Values.anchoreEnterpriseFeeds.githubDriverEnabled | quote) "true" }}
```

However that isn't the convention that is used extensively throughout the chart, so I'm not sure if we should introduce that in this single spot.

Reply:
In the previous chart, the values would not work if the booleans were not quoted. With this update, the chart works in either case. However, this check only works if the boolean is not quoted. I think that this may cause issues for customers who upgrade. If we do not change this, we should be very careful in communicating to customers that they should unquote the booleans in their values files.

Reply:
After thorough testing, I recommend using `{{- if eq (.Values.anchoreEnterpriseFeeds.githubDriverEnabled | toString) "true"}}`

Reply:
Nice find! `toString` is definitely the better solution here. We should put in an issue for updating all `if` statements that are expecting true/false values to follow this condition.